WBooks/AddNew/AddNewViewController.swift

```diff
@@ -23,11 +23,110 @@ class AddNewViewController: UIViewController {
+    @objc func selectImage(){
+        let alertController = UIAlertController(title: .none, message: .none, preferredStyle: .actionSheet)
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.delegate = self
+        let chooseAction = UIAlertAction(title: NSLocalizedString("OPTION_GALLERY", comment: ""), style: .default) { _ in
+            imagePickerController.sourceType = .photoLibrary
+            self.present(imagePickerController, animated: true, completion: .none)
+        }
+        alertController.addAction(chooseAction)
+        if UIImagePickerController.isSourceTypeAvailable(.camera) {
+            let takeAction = UIAlertAction(title: NSLocalizedString("OPTION_CAMERA", comment: ""), style: .default) { _ in
+                imagePickerController.sourceType = .camera
+                self.present(imagePickerController, animated: true, completion: .none)
+            }
+            alertController.addAction(takeAction)
+        }
+        let cancelAction = UIAlertAction(title: NSLocalizedString("OPTION_CANCEL", comment: ""), style: .cancel, handler: .none)
+        alertController.addAction(cancelAction)
+        self.present(alertController, animated: true, completion: nil)
+    }
+    
+    @objc func validateAddBook() {
+        if isEmpyForm() {
+            addNewViewModel.setParametersBook(
+                title: addNewView.bookNameTextField.text ?? "",
+                author: addNewView.authorTextField.text ?? "",
+                genre: addNewView.topicTextField.text ?? "",
+                year: addNewView.yearTextField.text ?? "")
+            addBook()
+        } else {
+            self.showAlert(title: NSLocalizedString("TITLE_ERROR", comment: ""),
+                           message: NSLocalizedString("ERROR_EMPY_TEXTFLIED", comment: ""),
+                           buttonDone: NSLocalizedString("BUTTON_DONE", comment: ""))
+        }
+    }
+    
+    func addBook()  {
+        addNewViewModel.addNewBook() { error in
+            if error == nil {
+                self.cleanForm()
+                self.showAlert(title: NSLocalizedString("TITLE_INFORMATION", comment: ""),
+                               message: NSLocalizedString("ADDBOOK_SUCCESS", comment: ""),
+                               buttonDone: NSLocalizedString("BUTTON_DONE", comment: ""))
+            } else {
+                self.cleanForm()
+                self.showAlert(title: NSLocalizedString("TITLE_ERROR", comment: ""),
+                               message: NSLocalizedString("ERROR_ADDBOOK", comment: ""),
+                               buttonDone: NSLocalizedString("BUTTON_DONE", comment: ""))
+            }
+        }
+    }
+    
+    func isEmpyForm() -> Bool {
+        return addNewView.bookNameTextField.isValidValue() && addNewView.authorTextField.isValidValue() && addNewView.yearTextField.isValidValue() && addNewView.topicTextField.isValidValue() && addNewView.descriptionTextField.isValidValue()
```

Comment:
If we have only one line, we can avoid the return

Reply:
Add a space after and operator to improve the readability

Reply:
Done